Server/uploadpackage.R

```diff
@@ -92,29 +83,28 @@ observe({
   rm(CRAN_avail_pkgs)
 
   
-  waitress$inc(2)
-  
   # Current packages on the db.
   curr_pkgs <- dbSelect("SELECT name FROM package")
   
-  waitress$inc(1)
-  
   # Save the uploaded packages that were not in the db.
   new_pkgs <- uploaded_packages %>% filter(!(package %in% curr_pkgs$name))
-  
+
+  withProgress(message = "Uploading Packages to DB:", detail = "go get some coffee!", value = 0, {
   if(nrow(new_pkgs) != 0){
-    for (pkg in new_pkgs$package) {
+    for (i in seq_along(new_pkgs$package)) {
+      pkg <- new_pkgs$package[i]
+      incProgress(1 / (nrow(new_pkgs) + 1), detail = paste("package", pkg, "version", new_pkgs$version[i]))
       # Get and upload pkg general info to db.
       insert_pkg_info_to_db(pkg)
-      waitress$inc(1)
       # Get and upload maintenance metrics to db.
       insert_maintenance_metrics_to_db(pkg)
-      waitress$inc(1)
       # Get and upload community metrics to db.
       insert_community_metrics_to_db(pkg)
-      waitress$inc(1)
     }
   }
+    setProgress(value = 1, detail = "-- **Uploading Packages Completed**")
+    Sys.sleep(0.25)
```

Comment:
Hi @Robert-Krajcik, is this `Sys.sleep(.25)` necessary? What is it doing?

Reply:
gives us 1/4 second to see the "uploading packages completed" message